### PR TITLE
Update Chapel release version number to 1.19.0.

### PR DESCRIPTION
**Pull request by jhoole**
Update from 1.18.0 to 1.19.0 and from 1.9.10 (pre-release) to 1.19.0 where applicable in preparation for the release.

### CODE REVIEW

**Comment by ronawho** [comment]
I'd look at previous versions of this PR for other files that need to be updated. e.g. https://github.com/chapel-lang/chapel/pull/11097

I'd also search for "pre-release" and probably 1.18.0 and/or 1.18 to see if there's any new files that need to be updated (most are probably .bad files and comments that don't need to be updated)

File: test/compflags/bradc/printstuff/versionhelp.sh [hunk 1]
```diff
@@ -5,4 +5,4 @@ compiler=$3
 echo -n `basename $compiler`
 cat $CWD/version.goodstart
 diff $CWD/../../../../compiler/main/BUILD_VERSION $CWD/zero.txt > /dev/null 2>&1 && echo "" || \
-    { echo -n " pre-release (" && cat $CWD/../../../../compiler/main/BUILD_VERSION | tr -d \"\\n && echo ")" ; }
+    { echo -n "."" && cat $CWD/../../../../compiler/main/BUILD_VERSION | tr -d \" ; }
```
Comment (bradcray) [inline test/compflags/bradc/printstuff/versionhelp.sh]: Looks like there's a stray double-quote here?

Comment (jhoole) [reply test/compflags/bradc/printstuff/versionhelp.sh]: Gah

**Comment by bradcray** [comment]
Other than the double double-quote I noted, this looks OK to me.  Things I did to sanity-check it were:

* building the release tarball and 'chpl' from that and verifying that the version was 1.19.0
* verifying that docs built from this branch or from the tarball are labeled as 1.19.0 docs
* grepped for other instances of pre*release, 1.18, etc. that looked worrisome and none did

I'd also like to run a paratest on this before cutting the 1.19 branch (and, optionally, before merging this), though I have another run currently testing.  I'll do this one next.

Two additional minor observations of note:

* conf.rst has a reference to 1.19 in a comment which looks like it's dead code at this point (?).  It might be worthwhile removing this to avoid worrying about it being out-of-date in the future?  (but I must admit to not being very familiar to 1.19)
* doc/rst/tools/mason/mason.rst has a bunch of references to Chapel 1.18 as part of its documentation.  On one hand, I'm inclined to leave them as-is because I don't like the idea of adding more things that need to be updated rather than reducing that list as much as possible (through scripting and automation); on the other hand, would it be something a user would trip over?  (I didn't read the docs to check).  Tagging @ben-albrecht who knows this documentation the best.

**Comment by ben-albrecht** [comment]
> conf.rst has a reference to 1.19 in a comment which looks like it's dead code at this point (?). It might be worthwhile removing this to avoid worrying about it being out-of-date in the future? (but I must admit to not being very familiar with this file)

I'm OK dropping this. This was one of the config definitions we dropped to remove the redundant version numbers that showed up when you posted a link to certain platforms, e.g. "1.19.0 Chapel Documentation 1.19"

> doc/rst/tools/mason/mason.rst has a bunch of references to Chapel 1.18 as part of its documentation. On one hand, I'm inclined to leave them as-is because I don't like the idea of adding more things that need to be updated rather than reducing that list as much as possible (through scripting and automation); on the other hand, would it be something a user would trip over? (I didn't read the docs to check).

I think it's OK to leave these as is. The version number will be inferred from the user's compiler by default, so I don't think it'd be too confusing. We could always add a note to the doc saying "This version number will be the version you are current working with" in the future.

**Comment by jhoole** [comment]
I may just be blind, but I'm having trouble locating conf.rst. What's the path to that?

**Comment by ben-albrecht** [comment]
> I may just be blind, but I'm having trouble locating conf.rst. What's the path to that?

That's because it's `conf.py` as in `doc/rst/conf.py`. Sorry for the confusion!

**Comment by bradcray** [comment]
Aargh, sorry, I mean conf.py.  Did I mention I was up late last night fixing bugs?

**Comment by jhoole** [comment]
And this is the section we're talking about taking out?
```
# The version info for the project you're documenting, acts as replacement for
# |version| and |release|, also used in various other places throughout the
# built documents.
#
# The short X.Y version.
# version = '1.19'

# We use a custom version variable (shortversion) instead, because setting
# 'version' adds a redundant version number onto the top of the sidebar
# automatically (rtd-theme). We also don't use |version| anywhere in rst

chplversion = '1.19'                  # TODO -- parse from `chpl --version`
shortversion = chplversion.replace('-', '&#8209') # prevent line-break at hyphen, if any
```

**Comment by bradcray** [comment]
Yes, I was anticipating the day that the line:

```
# version = '1.19'
```

would start causing hits for someone doing a grep to make sure all 1.19's had been converted to 1.20's

**Comment by jhoole** [comment]
Okay. We're just talking about taking the comment out, not the actual code functionality?

**Comment by bradcray** [comment]
Right.  And it need not be done now, it was just an observation while reviewing and looking for missed cases.

**Comment by bradcray** [comment]
(Oops, missed that it had already happened—great!)

**Comment by bradcray** [comment]
@jhoole: I verified that this passed testing (as expected).

**Pull request merged**